src/main/java/org/codehaus/mojo/versions/SetMojo.java

```diff
@@ -375,16 +370,20 @@ public void execute() throws MojoExecutionException, MojoFailureException
                         && oldVersionIdRegex.matcher( mVersion ).matches()
                         && !newVersion.equals( mVersion ) )
                 {
-                    found = true;
-                    // if the change is not one we have swept up already
                     applyChange( project, reactor, files, mGroupId, m.getArtifactId(),
                             StringUtils.isBlank( oldVersion ) || "*".equals( oldVersion ) ? "" : mVersion );
                 }
             }
-            if ( !found && RegexUtils.getWildcardScore( groupId ) == 0 && RegexUtils.getWildcardScore( artifactId ) == 0
-                && RegexUtils.getWildcardScore( oldVersion ) == 0 )
+
+            if ( "always".equals( updateBuildOutputTimestampPolicy ) )
             {
-                applyChange( project, reactor, files, groupId, artifactId, oldVersion );
+                reactor.values().parallelStream()
```

Comment:
What benefit  we have using `parallelStream`?

Reply:
allows the operation to be performed in parallel (`stream` forces single-threaded execution) and here we don't depend on the order of execution